R/preprocess.R

```diff
@@ -979,14 +972,16 @@ enw_construct_data <- function(obs, new_confirm, latest, missing_reference,
 #' pobs
 enw_preprocess_data <- function(obs, by = NULL, max_delay = 20,
                                 set_negatives_to_zero = TRUE,
-                                ...) {
-  obs <- check_dates(obs)
+                                ..., copy = TRUE) {
+  # coerce obs - at this point, either making a copy or not
+  # after, we are modifying the copy/not copy
+  obs <- coerce_dt(obs, dates = TRUE, copy = copy)
```

Comment:
check here for `confirm` which is currently a hard requirement for preprocessing.

Reply:
counter-offer: enforce it in `enw_add_max_reported`, where it first appears as a requirement.